### DISCUSSION TITLE
How to add dark/light mode to my next.js blog?

### DISCUSSION THREAD

**Discussion by SachinMaharana** [vercel/next.js#12171] [Help] [answered]
I have started doing som web dev for my personal  project and figuring out how to implmenet a dark mode for the next.js with styled-jsx. i haven't come across any exmaples. Any idea help me figure out this would be of great help.

**Comment by erikdstock** [comment]
since styled-jsx uses interpolated css values you may be able to use the react state api to inject some main body/text colors at a high level. You may also have luck with a theming library that supports color modes- [theme-ui](https://theme-ui.com) which has this and a next.js example in its repository.

**Comment by leerob** [comment]
Here's an example using `styled-jsx` that has dark mode.

- https://static-tweet.now.sh/
- https://github.com/lfades/static-tweet

**Comment by kachkaev** [comment, marked as answer]
Here is what I did in [njt](https://njt.now.sh/) (npm jump to): https://github.com/kachkaev/njt/commit/1ef3bb026b006a355173c8a55594de65de53c102

* Uses https://github.com/donavon/use-dark-mode
* Based on setting `.dark-mode` / `.light-mode` to body. I'm using these extra class names in styled-component styles, but the trick should also work in any other styling solution
* No initial page flickering because of a raw js injection into `_document.tsx`
* Currently no on/off switch, only relying on OS values. But:
  * Adding an on/off switch should be easy, hence `useDarkMode()` is already used in `_app.tsx`
  * Changing dark mode in macOS changes the mode of the page without page reload 👀

![dark mode small](https://user-images.githubusercontent.com/608862/82750753-23de6500-9daa-11ea-970b-32b6b20b08c7.gif)
